tutorial/managing-software.rst

```diff
@@ -0,0 +1,1611 @@
+    bsdutils/noble-updates 1:2.39.3-9ubuntu6.1 amd64 [upgradable from: 1:2.39.3-9ubuntu6]
+    cloud-init/noble-updates 24.2-0ubuntu1~24.04.2 all [upgradable from: 24.1.3-0ubuntu3.3]
+    [...]
+
+apt upgrade
+-----------
+
+The ``apt upgrade`` command is about the **packages** on your system. It looks at
+the metadata in the package index we just updated, finds the packages with
+available upgrades, and lists them for us.
+
+After we have updated the database (which we did by running ``apt update``) we can
+then upgrade the packages to their newest versions by running:
+
+.. code-block:: bash
+
+    sudo apt upgrade
+
+When we run this command, it will ask us to confirm if the summary of proposed
+changes that will be made to our system is what we want.
+
+Let's type :kbd:`Y`, then press :kbd:`Enter` to confirm that yes, we do
+want that, and then the upgrade will proceed. This may take a few minutes.
+
+.. tip::
+   You can use the ``-y`` flag, which is a shorthand for ``--assume-yes``. If we ran
+   the command ``sudo apt upgrade -y`` it would proceed with the upgrade without
+   asking us to confirm. Shorthand versions of flags are common -- for most
+   packages, you can check which flags are equivalent using
+   `the manual pages <https://manpages.ubuntu.com/>`_ or using the ``man``
+   command, as we'll see later. 
+
+In the output, we'll see where ``apt upgrade`` is fetching the upgrade from for
+each package. For example:
+
+.. code-block:: text
+
+    Get:1 http://archive.ubuntu.com/ubuntu noble-updates/main amd64 libopeniscsiusr amd64 2.1.9-3ubuntu5.1 [49.1 kB]
+
+APT combines the various elements; the package name (``libopeniscsiusr``),
+version (``2.1.9-3ubuntu5.1``), source (``noble-updates/main``), etc into a single
+URL that it can use for the download. It then unpacks the package and applies
+the upgrade to the system.
+
+.. note::
+    These commands only upgrade the packages for the release of Ubuntu that we are
+    using (24.04 LTS). If we wanted to upgrade to the next release of Ubuntu, we
+    would use the ``do-release-upgrade`` command. See this guide on
+    :ref:`how to upgrade your release <upgrade-your-release>` for more information.
+
+It's important to know that ``apt upgrade`` will only handle packages that can be
+straightforwardly upgraded. If the package has **dependency** issues (i.e., the
+version you have "depends" on other packages that also need to be added,
+upgraded or removed), you would need to use ``sudo apt dist-upgrade`` instead.
+The ``dist-upgrade`` command is able to resolve conflicts between package
+versions, but it *could* end up removing some packages -- so although
+``apt upgrade`` is safe to use unattended (in a script, for example), you should
+only use ``dist-upgrade`` when you can pay attention to it.
+
+Searching APT
+-------------
+
+Now we're up-to-date, we can start exploring! As with any other database, we
+can search APT to find software. Let's say that we want to find a webserver,
```

Comment:
Since we're using APT to search, I've updated the section heading to "Searching with APT" (hopefully accurate enough to reflect what you're suggesting here) and updated the text with your suggested wording.